src/Jackett.Common/Indexers/FileList.cs

```diff
@@ -99,109 +100,169 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
+            if (imdbID != null)
+            {
+                queryCollection.Add("search", imdbID);
+            }
             else if (!string.IsNullOrWhiteSpace(searchString))
+            {
                 queryCollection.Add("search", searchString);
+            }
 
             queryCollection.Add("cat", cat);
             queryCollection.Add("searchin", "1");
             queryCollection.Add("sort", "2");
+            queryCollection.Add("page", page.ToString());
 
-            searchUrl += "?" + queryCollection.GetQueryString();
+            return queryCollection;
+        }
 
-            var response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
+        private async Task<string> callProviderAsync(TorznabQuery query, int page)
+        {
+            var searchUrl = BrowseUrl;
+            var searchString = query.GetQueryString();
+
+            var cat = MapTorznabCapsToTrackers(query).FirstIfSingleOrDefault("0");
+
+            searchUrl += "?" + getQueryCollection(query.ImdbID, searchString, cat, page).GetQueryString();
 
+            var response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
             // Occasionally the cookies become invalid, login again if that happens
             if (response.IsRedirect)
             {
                 await ApplyConfiguration(null);
                 response = await RequestStringWithCookiesAndRetry(searchUrl, null, BrowseUrl);
             }
 
-            var results = response.Content;
+            return response.Content;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var pageCount = 0;
+            var firstResponse = await callProviderAsync(query, 0);
+
+            pageCount = getPageCount(firstResponse);
+            transformProviderRequest(query, firstResponse, releases);
+            //Wait
+            Thread.Sleep(50);
+
+            //Process the rest of the pages
+            if (pageCount > 0 && query.GetQueryString() != "")
+            {
+                for (int i = 1; i < pageCount; i++)
+                {
+                    var resp = await callProviderAsync(query, i);
+                    transformProviderRequest(query, resp, releases);
+                    //Wait between requests
+                    Thread.Sleep(50);
+                }
+
+            }
+
+            return releases;
+        }
+
+        private Int16 getPageCount(String html)
+        {
+            try
+            {
+                var parser = new HtmlParser();
+                var dom = parser.ParseDocument(html);
+                var pageCount = dom.QuerySelectorAll(".pager > span > a > font").Last().InnerHtml;
+                return Int16.Parse(pageCount);
+            }
+            catch (Exception)
+            {
+                logger.Info("Only one page found.");
+            }
```

Comment:
Catching `Exception` is discouraged as it can suppress errors you weren't intending to handle, and catching as flow control is also discouraged. This would be better served as an if statement. Something like the following:
```C#
var pageCount = dom.QuerySelectorAll(".pager > span > a > font");
if(!pageCount.Any())
{
    logger.Info("Only one page found.");
    return 0;
}
return int.Parse(pageCount.Last().TextContent);
```

Reply:
What do you mean with:  
"Although not specifically mentioned in the style guidelines, please also inline any private functions you've made that are only used once."
?